pkg/webhook/admission/http.go

```diff
@@ -34,6 +34,9 @@ import (
 var admissionScheme = runtime.NewScheme()
 var admissionCodecs = serializer.NewCodecFactory(admissionScheme)
 
+// based on https://github.com/kubernetes/kubernetes/blob/c28c2009181fcc44c5f6b47e10e62dacf53e4da0/staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go
+var maxRequestSize = int64(3 * 1024 * 1024)
```

Comment:
This is technically a breaking change. More importantly, how do we know this size is appropriate? I guess for admission we can inferr that a request contains at most two objects which each have a maximum size of one meg plus the request itself which is also not likely to exceed one meg. But how about requests in the authentication webhook?

Reply:
Upstream, this value seems to come from here: https://github.com/kubernetes/apiserver/blob/master/pkg/server/config.go#L433-L442C5

Reply:
So, k/k has made a mistake using that value for this admission webhook?: https://github.com/kubernetes/kubernetes/blob/c28c2009181fcc44c5f6b47e10e62dacf53e4da0/staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go

Reply:
@alvaroaleman I updated the maxRequestSize based on the reasoning you explained and added comments: https://github.com/kubernetes-sigs/controller-runtime/pull/2598/commits/9dcfc53b6dd7852c484034b763b015258dd382a6